ideascube/widgets.py

```diff
@@ -1,6 +1,56 @@
 from django.forms import widgets
 
 from ideascube.configuration import get_config
+from django.utils.html import mark_safe, escape
+from django.utils.encoding import force_text
+from django.forms.utils import flatatt
+
+
+_script_template = '''<script type="text/javascript">tinymce.init({{
+selector: "#{widget_id}",
+inline: true,
+menubar: {menubar},
+toolbar: {toolbar},
+plugins: "{plugins}",
+hidden_input: false
+}});</script>'''
+
+
+class RichTextEntry(widgets.Widget):
+    def __init__(
+        self, menubar=False, toolbar=False, plugins="autolink", attrs=None):
+        super().__init__(attrs=attrs)
+        self.menubar = menubar
+        self.toolbar = toolbar
+        self.plugins = plugins
+
+    def render(self, name, value, attrs=None):
+        if value is None:
+            value = ''
+
+        # The widget displayed to the user
+        div_attrs = {'class': 'tinymce_editable',
+                        'id': name}
+        div_html = '<div{}>{}</div>'.format(
+            flatatt(div_attrs),
+            force_text(value)
+        )
+
+        # The input used by django
+        input_attrs = self.build_attrs(attrs, name=name, value=escape(value), type='hidden')
+        input_html = '<input{} />'.format(
+            flatatt(input_attrs)
+        )
+
+        # Make tinymce handle the widget
+        script_html = _script_template.format(
+            widget_id = name,
+            menubar = 'true' if self.menubar else 'false',
+            toolbar = 'true' if self.toolbar else 'false',
+            plugins = self.plugins)
```

Comment:
I'm not sure I like adding the `<script>` this way.

Why not simply something like the following, at the bottom of `form-fullpage.html`:

```javascript
...
```

Hmm, I see the problem: using `selector: ".tinymce_editable"` as an argument of `tinymce.init()` means that all widgets have the same `toolbar`/`menubar`/`plugins`. :-/

Maybe something like this?

```javascript
var items = document.querySelectorAll('.tinymce_editable');

for (var i = 0; i < items.length; i++) {
    var item = items[i];

    tinymce.init({
        selector: '#' + item.getAttribute('id'),
        inline: true,
        menubar: item.getAttribute('data-menubar'),
        toolbar: item.getAttribute('data-toolbar'),
        plugins: item.getAttribute('data-plugins'),
        hidden_input: false
    });
}
```

That avoids scattering potentially lots of `<script>` on the page, all items get handled at the same place, and it removes the inelegant templating.

But then maybe it's too complicated to get it working within the constraints of the `data-*` attributes?